daemon/containerd/image_tag.go

```diff
@@ -81,15 +81,13 @@ func (i *ImageService) createOrReplaceImage(ctx context.Context, newImg c8dimage
 	logger.Info("image created")
 
 	if !creatingDangling {
-		ctx := context.WithoutCancel(ctx)
-		defer i.LogImageEvent(string(newImg.Target.Digest), imageFamiliarName(newImg), events.ActionTag)
-
 		if err := i.images.Delete(ctx, danglingName); err != nil {
 			if !cerrdefs.IsNotFound(err) {
 				logger.WithError(err).Warn("unexpected error when deleting dangling image")
 			}
 		}
 	}
 
+	i.LogImageEvent(ctx, string(newImg.Target.Digest), imageFamiliarName(newImg), events.ActionTag)
```

Comment:
Double check if this change is still correct now that the `if !creatingDangling {` condition was added

Reply:
No, it should only be logged  if `!creatingDangling`. Otherwise a `tag` event will be emitted when a dangling image is created (in this case, only `create` should be emitted.

Reply:
Ah, thx! I'll update. I confused myself for a bit when looking at the diff again after rebasing.